softioc/device.py

```diff
@@ -473,6 +473,9 @@ def _epics_to_value(self, value):
         if self._dtype.char == 'S':
             return [_string_at(s, 40) for s in value]
         else:
+            # As we return a reference to the numpy array, ensure it cannot be
+            # modified under our noses
+            value.flags.writeable = False
```

Comment:
Probably better to do this on the _incoming_value, so do this a few lines above here https://github.com/dls-controls/pythonSoftIOC/blob/8e555c2b95bd3d0c3f8c05b8b9fbd2ea4a1db27e/softioc/device.py#L470 before returning the copy.

Reply:
Done in  8c4b516